From cc34f1fa6be5223ef14ef5dbc5717cfbe83239bb Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@google.com>
Date: Fri, 20 Nov 2020 14:54:15 +0100
Subject: [PATCH 807/830] ANDROID: GKI: Enable CONFIG_USB_TYPEC_UCSI
X-NVConfidentiality: public

GKI 1 had CONFIG_USB_TYPEC and CONFIG_USB_TYPEC_TCPM enabled, but forgot
to enable the _UCSI driver as well.  Do that now as everyone relies on
this class driver.

If platform-specific TYPEC changes are needed, just enable the
platform-specific tyepc driver for the specific platform.

Note, for some reason x86 forgot to enable TYPEC entirely, so fix that
up and enable it here so we catch build issues sooner.

Bug: 157965270
Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
Change-Id: I353eec6ed4f73525408033f283081b8013ede874
---
 arch/arm64/configs/gki_defconfig | 1 +
 arch/x86/configs/gki_defconfig   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index aaa33f71a277..cf4ebc6adf99 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -434,6 +434,7 @@ CONFIG_USB_CONFIGFS_F_HID=y
 CONFIG_TYPEC=y
 CONFIG_TYPEC_TCPM=y
 CONFIG_TYPEC_TCPCI=y
+CONFIG_TYPEC_UCSI=y
 CONFIG_MMC=y
 # CONFIG_PWRSEQ_EMMC is not set
 # CONFIG_PWRSEQ_SIMPLE is not set
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index fe8ad763b618..c57654a6ab63 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -389,6 +389,7 @@ CONFIG_USB_CONFIGFS_F_HID=y
 CONFIG_TYPEC=y
 CONFIG_TYPEC_TCPM=y
 CONFIG_TYPEC_TCPCI=y
+CONFIG_TYPEC_UCSI=y
 CONFIG_MMC=y
 # CONFIG_PWRSEQ_EMMC is not set
 # CONFIG_PWRSEQ_SIMPLE is not set
-- 
2.17.1

